libfwupdplugin/fu-config.c

```diff
@@ -492,17 +517,26 @@ fu_config_add_location(FuConfig *self, const gchar *dirname, GError **error)
 
 	/* is writable */
 	if (g_file_query_exists(item->file, NULL)) {
-		g_autoptr(GFileInfo) info = g_file_query_info(item->file,
-							      G_FILE_ATTRIBUTE_ACCESS_CAN_WRITE,
-							      G_FILE_QUERY_INFO_NONE,
-							      NULL,
-							      error);
+		g_autoptr(GFileInfo) info = NULL;
+
+		g_debug("loading config %s", item->filename);
+		info = g_file_query_info(item->file,
+					 G_FILE_ATTRIBUTE_ACCESS_CAN_WRITE,
+					 G_FILE_QUERY_INFO_NONE,
+					 NULL,
+					 error);
 		if (info == NULL)
 			return FALSE;
 		item->is_writable =
 		    g_file_info_get_attribute_boolean(info, G_FILE_ATTRIBUTE_ACCESS_CAN_WRITE);
+		if (item->is_writable) {
+			if (g_str_has_suffix(dirname, "tests/fwupd"))
+				item->is_writable = FALSE;
+		}
```

Comment:
How about in the tests marking it immutable instead?  Then less special test code in `fu-config.c`

Reply:
> How about in the tests marking it immutable instead

You mean a g_setenv or something?

Reply:
No; I mean literally changing the Linux immutable attribute of the file in the test when it's created.

Reply:
Rather than use the file that is "in the git tree", make a copy of it at the start of the test and make the _copy_ immutable

Reply:
Or another idea; just mark the in-tree file as read-only.  That should accomplish the same result as a linux immutable file.

Reply:
Good idea, lets try that in CI.

Reply:
Ahh, so it works locally, until you clone a new checkout and then it's r/w -- I'll put my kludge back to fix CI.

Reply:
Just remove the file before starting instead?

Reply:
Nah, the test is explicitly checking a pre-populated non-mutable file.